data_validation/combiner.py

```diff
@@ -34,27 +36,106 @@
     from data_validation.metadata import RunMetadata, ValidationMetadata
 
 
+# At around 140 columns we hit RecursionError when unioning Ibis subqueries.
+# This constant is a threshold at which we slice up the input Dataframes
+# and stitch them back together again after Ibis processing.
+COMBINER_COLUMN_SLICE_WIDTH = 120
+
 COMBINER_GET_SUMMARY_EXC_TEXT = (
     "Error while generating summary report of row validation results"
 )
 
 
 def generate_report(
-    client,
     run_metadata: "RunMetadata",
-    source: "IbisTable",
-    target: "IbisTable",
+    source_df: pandas.DataFrame,
+    target_df: pandas.DataFrame,
```

Comment:
```suggestion
    source_df: "DataFrame",
    target_df: "DataFrame",
```
Shouldn't we use the same type hint approach as used for the other params? `DataFrame` is already imported on L34 specifically in the `TYPE_CHECKING` block

Reply:
You are right to flag this inconsistency. Because we need to import pandas for `pandas.concat` we now have the package imported and can use that instead of having a second import for typing only. So I was thinking to replace all `"DataFrame"` with `pandas.DataFrame` but then forgot to come back to it.

I don't suppose it really matters if we have a second import really. I'll leave this comment open for now and come back to it.